### ISSUE TITLE
Nested notes with `split` and `reset`

### ISSUE THREAD

**Issue by gusbrs** [cgnieder/enotez#38]
In playing with nested notes to understand better its implications for #37, I came up with an example of interaction between `split`, `reset` and the use of nested notes which is unexpected, or at least unfortunate.

Consider the document:

```latex
\documentclass{book}

\usepackage{enotez}
\setenotez{
  split=chapter,
  reset,
}

\begin{document}

\chapter{Chapter 1}

Text\endnote{Foo.\endnote{Foo nested.}}

Text\endnote{Bar.\endnote{Bar nested.\endnote{Bar nested nested.}}}

\chapter{Chapter 2}

Text\endnote{Baz.\endnote{Baz nested.}}

\printendnotes

\end{document}
```
It results in the following endnotes:

![Screenshot from 2022-03-19 08-34-00](https://user-images.githubusercontent.com/19410606/159119472-d5912dfc-9042-4f0f-a642-99405a52c8fe.png)

As you can see, a number of problems occur:

- Nested notes are not kept in the same section as their "parent" notes, but are grouped into another section.
- This "nested notes" section gathers nested notes from *all* previous sections of non-nested notes.
- The double nested notes behave differently, and are kept within the same section as the one-level nested.
- The split-title takes the number of whatever the last numbered chapter was, so we got a repeated "last split section".

Of course, it is easy to understand what is happening here. This all derives from resetting the counter at the start of `\printendnotes`, since this does not happen at the split header, all nested notes are grouped into a single section. The split-title is tripped because the chapter at the start of "Notes" is unnumbered, the last value of a numbered section is what gets to prevail.

**Comment by cgnieder** [comment]
I think this is a duplicate of issue #16.

The only actual problem I see is that the third heading says “Notes for chapter 2”. All those nested notes _do_ stem from the “Notes” chapter, even the double nested ones. IMHO the grouping is correct but the naming of the third split section title is not but would be with `list-heading=\chapter`.

**Comment by cgnieder** [comment]
Something like this might be a solution:

```latex
\documentclass{book}

\usepackage{enotez}
\setenotez{
  split=chapter,
  reset
}

\newcommand*\lasttitle{}
\AtEveryListSplit{
  \ifx\lasttitle\EnotezCurrentSplitTitle
    \def\EnotezCurrentSplitTitle{Notes from this chapter}
  \fi
  \let\lasttitle\EnotezCurrentSplitTitle
}

\begin{document}

\chapter{Chapter 1}

Text\endnote{Foo.\endnote{Foo nested.}}

Text\endnote{Bar.\endnote{Bar nested.\endnote{Bar nested nested.}}}

\chapter{Chapter 2}

Text\endnote{Baz.\endnote{Baz nested.}}

\printendnotes

\end{document}
```

**Comment by gusbrs** [comment]
> The only actual problem I see is that the third heading says “Notes for chapter 2”. All those nested notes do stem from the “Notes” chapter, even the double nested ones. IMHO the grouping is correct but the naming of the third split section title is not but would be with `list-heading=\chapter`.

You are right. I was thinking in terms of "position in the source", but indeed the typeset marks from these notes do occur in the "Notes" section itself. It makes sense.

But, the split title is indeed unfortunate. And sure, using `list-heading=\chapter` would make the technical problem vanish. But I'd be surprised if people would want do decide on whether their "Notes" should be numbered or not for this reason, and that many would want a numbered chapter for it. Your idea for it is interesting, though it may be tripped if the user employs an unnumbered chapter/section of their own with an endnote in it (true, as things are, numbering is very much presumed for `split`).

And, as a matter of fact, the organization of nested notes "at the end" of `\printendnotes` (split or not) simplifies the problem of nested notes for the running headers at #37 , since this transition would occur only once. And a technique similar to the one you are suggesting here for the split title might be employed to "signal" this transition (at least if split is employed).

**Comment by cgnieder** [comment]
Actually, there should be an option to set the split title for individual chapters. I need to think about this…

**Comment by gusbrs** [comment]
That would be ideal. But probably a difficult UI case, if unnumbered chapters/sections are to be handled.

I remain at your disposal to discuss or brainstorm on this, if you wish.
